cluster-provision/images/vm-image-builder/README.md

```diff
@@ -0,0 +1,170 @@
+  You can use any image customizing script by exporting 'CUSTOMIZE_IMAGE_SCRIPT=<script path>'
+
+- Building container-disk image using `build-containerdisk.sh`
+  according to the doc: 
+  https://github.com/kubevirt/kubevirt/blob/master/docs/container-register-disks.md
+
+- Store the image in local registry and exports the container image to .tar file, so it will be easier to store or send.
+
+This script expected environment variables:
+VM_IMAGE_URL - URL to download the image
+CLOUD_CONFIG_PATH - cloud-conifg file path used  for customizing the image with cloud-init
+IMAGE_NAME, TAG - container-disk image tag that will be created tag details IMAGE_NAME:TAG
+OS_VARIANT- required by customize-image.sh
+
+Notes:
+Currently, image customization process done by spinning up live VM with cloud-init disk to provision the image with the changes described by the cloud-config file.
+
+It is necessary to add `sudo shutdown -h now` at the end of the cloud-config
+`runcmd` block, so the script will not wait for user input.
+
+You can use your own image customizing script by exporting the path to CUSTOMIZE_IMAGE_SCRIPT.
+
+This script requires the following packages:
+- cloud-utils
+- docker-ce
+- libvirt
+- libguestfs
+- qemu-img
+
+## How to use:
+```bash
+
+# Create kubernetes cluster with Kubvirt
+git clone https://github.com/kubevirt/kubevirt --depth=1 --branch=master --single-branch kubevirt
+$ pushd kubevirt
+$ export KUBEVIRT_PROVIDER=k8s-1.17
+$ make cluster-up cluster-sync
+$ popd
+
+# Optional, You could point to local cluster registry for developent and tests.
+# From kubevirt / kubevirtci directory
+$ export REGISTRY="localhost:$(./cluster-up/cli.sh ports registry | tr -d '\r')"
+
+# Set to which registry the image will be pushed
+$ export REGISTRY=docker.io
+$ export REPOSITORY=kubevirt
+
+# Export image name and tag
+$ export IMAGE_NAME="example"
+$ export TAG="tests"
+
+cd cluster-provision/images/container-disk-images/
+# Export VM image URL:
+$ export IMAGE_URL=$(cat example/example-image-url)
+$ export OS_VARIANT=fedora31
+
+# Export cloud-config file
+$ export CLOUD_CONFIG_PATH="example/example-cloud-config"
+ 
+# Run script
+$ ./create-containerdisk.sh
+
+# Create VM from the new image (requiers KubeVirt operator)
+$ export VMI_NAME="testvm1"
+$ export VMI_IMAGE="registry:5000/${REPOSITORY}/${IMAGE_NAME}:${TAG}"
+$ VMI_YAML="vmi-test-image.yaml"
+$ envsubst < "$VMI_YAML" | kubectl apply -f -
+
+$ kubectl wait --for=condition=AgentConnected vmi $VMI_NAME --timeout 5m
+$ virtctl console testvm1
+...
+
+# You could send / store the image .tar file: 
+$ ls example_build/example-tests.tar
+
+```
+
+## Build container-disk image
+
+To build container-disk image form qcow2 image file, use `build-containerdisk.sh` script.
+
+This script converts qcow2 image to container-disk image that can be consumed by KubeVirt VM's.
+
+What this script does:
+- Creates temp directory and copy the source
+  VM image and '
```

Comment:
`and '` ?

Reply:
Fixed
I also arranged this file and removed redundant explanations